From 7fd9ae8dfca591f5f0bda7850877c2f4fd38f3ec Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@gmail.com>
Date: Sun, 31 May 2026 14:47:15 +0100
Subject: [PATCH] fix(systemd-*): add new dlopen dependencies to modules lists

systemd v261 moved more dependencies to dlopen, including libssl which
is required by various components, like resolved
---
 modules.d/11systemd-creds/module-setup.sh       | 5 +++++
 modules.d/11systemd-pcrextend/module-setup.sh   | 5 +++++
 modules.d/11systemd-repart/module-setup.sh      | 2 +-
 modules.d/11systemd-resolved/module-setup.sh    | 5 +++++
 modules.d/11systemd-veritysetup/module-setup.sh | 2 +-
 modules.d/45systemd-import/module-setup.sh      | 2 +-
 modules.d/71systemd-cryptsetup/module-setup.sh  | 2 +-
 7 files changed, 19 insertions(+), 4 deletions(-)

diff --git a/modules.d/11systemd-creds/module-setup.sh b/modules.d/11systemd-creds/module-setup.sh
index 1b66519c31..642ff037d6 100755
--- a/modules.d/11systemd-creds/module-setup.sh
+++ b/modules.d/11systemd-creds/module-setup.sh
@@ -27,6 +27,11 @@ depends() {
 
 }
 
+# called by dracut
+config() {
+    add_dlopen_features+=" libsystemd-shared-*.so:libcrypto "
+}
+
 # Install the required file(s) and directories for the module in the initramfs.
 install() {
 
diff --git a/modules.d/11systemd-pcrextend/module-setup.sh b/modules.d/11systemd-pcrextend/module-setup.sh
index 8ebd9cf026..8c3dedcc92 100755
--- a/modules.d/11systemd-pcrextend/module-setup.sh
+++ b/modules.d/11systemd-pcrextend/module-setup.sh
@@ -34,6 +34,11 @@ depends() {
     return 0
 }
 
+# called by dracut
+config() {
+    add_dlopen_features+=" libsystemd-shared-*.so:libcrypto "
+}
+
 # Install the required file(s) and directories for the module in the initramfs.
 install() {
     inst_multiple -o \
diff --git a/modules.d/11systemd-repart/module-setup.sh b/modules.d/11systemd-repart/module-setup.sh
index e27799be28..931c753e67 100755
--- a/modules.d/11systemd-repart/module-setup.sh
+++ b/modules.d/11systemd-repart/module-setup.sh
@@ -13,7 +13,7 @@ check() {
 
 # Config adjustments before installing anything.
 config() {
-    add_dlopen_features+=" libsystemd-shared-*.so:fdisk "
+    add_dlopen_features+=" libsystemd-shared-*.so:fdisk,libcrypto "
 }
 
 # Install the required file(s) for the module in the initramfs.
diff --git a/modules.d/11systemd-resolved/module-setup.sh b/modules.d/11systemd-resolved/module-setup.sh
index 51e960cbe1..853f3cb480 100755
--- a/modules.d/11systemd-resolved/module-setup.sh
+++ b/modules.d/11systemd-resolved/module-setup.sh
@@ -26,6 +26,11 @@ depends() {
 
 }
 
+# called by dracut
+config() {
+    add_dlopen_features+=" libsystemd-shared-*.so:libcrypto,libssl "
+}
+
 # Install the required file(s) and directories for the module in the initramfs.
 install() {
 
diff --git a/modules.d/11systemd-veritysetup/module-setup.sh b/modules.d/11systemd-veritysetup/module-setup.sh
index 856926f6ea..af62c28416 100755
--- a/modules.d/11systemd-veritysetup/module-setup.sh
+++ b/modules.d/11systemd-veritysetup/module-setup.sh
@@ -28,7 +28,7 @@ depends() {
 
 # Config adjustments before installing anything.
 config() {
-    add_dlopen_features+=" libsystemd-shared-*.so:cryptsetup "
+    add_dlopen_features+=" libsystemd-shared-*.so:cryptsetup,libcrypto "
 }
 
 # Install kernel module(s).
diff --git a/modules.d/45systemd-import/module-setup.sh b/modules.d/45systemd-import/module-setup.sh
index daf3c9d415..abc70896bb 100755
--- a/modules.d/45systemd-import/module-setup.sh
+++ b/modules.d/45systemd-import/module-setup.sh
@@ -16,7 +16,7 @@ depends() {
 }
 
 config() {
-    add_dlopen_features+=" libsystemd-shared-*.so:archive,lz4,lzma,zstd "
+    add_dlopen_features+=" libsystemd-shared-*.so:archive,curl,libcrypto,lz4,lzma,zstd "
 }
 
 installkernel() {
diff --git a/modules.d/71systemd-cryptsetup/module-setup.sh b/modules.d/71systemd-cryptsetup/module-setup.sh
index 91ff96954e..65d729e60c 100755
--- a/modules.d/71systemd-cryptsetup/module-setup.sh
+++ b/modules.d/71systemd-cryptsetup/module-setup.sh
@@ -47,7 +47,7 @@ depends() {
 
 # called by dracut
 config() {
-    add_dlopen_features+=" libsystemd-shared-*.so:cryptsetup "
+    add_dlopen_features+=" libsystemd-shared-*.so:cryptsetup,libcrypto "
 }
 
 # called by dracut
